rclcpp/src/rclcpp/utilities.cpp

```diff
@@ -224,10 +224,19 @@ rclcpp::remove_ros_arguments(int argc, char const * const argv[])
     &nonros_argv);
 
   if (RCL_RET_OK != ret) {
+    // Not using throw_from_rcl_error, because we may need to append deallocation failures.
+    exceptions::RCLErrorBase base_exec(ret, rcl_get_error_state());
+    rcl_reset_error();
     if (NULL != nonros_argv) {
       alloc.deallocate(nonros_argv, alloc.state);
     }
-    exceptions::throw_from_rcl_error(ret, "Failed to remove ROS arguments: ");
+    if (RCL_RET_OK != rcl_arguments_fini(&parsed_args)) {
+      base_exec.formatted_message += std::string(
+        ", failed also to cleanup parsed arguments, leaking memory: ") +
+        rcl_get_error_string_safe();
+      rcl_reset_error();
+    }
+    throw base_exec;
```

Comment:
I think our default behaviour is to throw an `RCLError` not an `RCLErrorBase`. I would change this to
```
throw RCLError(base_exc, "");
```
following 
 https://github.com/ros2/rclcpp/blob/c70f2f145239ca4dfe86c6d806f331602a0e5c68/rclcpp/src/rclcpp/exceptions.cpp#L71..L72